.github/workflows/weekly.yml

```diff
@@ -19,6 +19,11 @@ jobs:
       - name: Install Rust
         uses: dtolnay/rust-toolchain@stable
 
+      - name: Set up Cargo Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          save-if: true
+
```

Comment:
So let me see if I'm following it correctly. The idea is to keep the weekly job, but adding a new workflow that has
```
on:
  push:
    branches:
      - main
``` as predicate instead of a weekly schedule, right?

Reply:
Yup, the Weekly will remain same as we have in `main` right now. Cache will be created on pushing or pull request to `main`.

```yml
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```

Reply:
Perfect :+1: got it

Reply:
I think caching on push or PR to the main branch will be sufficient. Therefore, we should remove caching from the weekly workflow.
```yml
        with:
          save-if: false
```